respy/likelihood.py

```diff
@@ -346,3 +400,41 @@ def _process_estimation_data(df, options):
     df = _convert_choice_variables_from_categorical_to_codes(df, options)
 
     return df
+
+
+def _adjust_options_for_estimation(options, df):
+    """Adjust options for estimation.
+
+    There are some option values which are necessary for the simulation, but they can be
+    directly inferred from the data for estimation. A warning is raised for the user
+    which can be suppressed by adjusting the options.
+
+    """
+    for choice in options["choices_w_exp"]:
+
+        # Adjust initial experience levels for all choices with experiences.
+        init_exp_data = np.sort(
+            df.loc[df.Period.eq(0), f"Experience_{choice.title()}"].unique()
+        )
+        init_exp_options = options["choices"][choice]["start"]
+        if not np.array_equal(init_exp_data, init_exp_options):
+            warnings.warn(
+                f"The initial experience for choice '{choice}' differ between data, "
+                f"{init_exp_data}, and options, {init_exp_options}. The options are "
+                "adjusted.",
+                category=UserWarning,
+            )
+            options["choices"][choice]["start"] = init_exp_data
+            options["choices"][choice].pop("share")
+            options["choices"][choice].pop("lagged")
+
+        # Adjust the number of periods.
+        if not options["n_periods"] == df.Period.max() + 1:
+            warnings.warn(
+                f"The number of periods differ between data, {df.Period.max()}, and "
+                f"options, {options['n_periods']}. The options are adjusted.",
```

Comment:
Maybe the message should be that those options are ignored instead of adjusted because we don't really modify them anymore.

Reply:
👍